### PR TITLE
Adding evaluation pipeline to BERT examples

### PR DESCRIPTION
**Pull request by yzhang123**
* adding evaluation pipeline to BERT examples
* renamed Squad input argument `infer_file` to `test_file`

### CODE REVIEW

File: docs/sources/source/nlp/bert_pretraining.rst [hunk 1]
```diff
@@ -191,7 +192,8 @@ For training from raw text use nemo_nlp.nm.data_layers.BertPretrainingDataLayer,
             # data_layer = nemo_nlp.BertPretrainingPreprocessedDataLayer(
```
Comment (ekmb) [inline docs/sources/source/nlp/bert_pretraining.rst]: should it be nemo_nlp.nm.data_layers.BertPretrainingPreprocessedDataLayer ?

File: docs/sources/source/nlp/bert_pretraining.rst [hunk 2]
```diff
@@ -25,7 +25,7 @@ Download Corpus
 The training corpus can be either raw text where data preprocessing is done on the fly or an already preprocessed data set. In the following we will give examples for both.
 To showcase how to train on raw text data, we will be using the very small WikiText-2 dataset :cite:`nlp-bert-merity2016pointer`.
 
-To download the dataset, run the script ``examples/nlp/scripts/get_wt2.sh``. After downloading and unzipping, the folder should include 3 files that look like this:
+To download the dataset, run the script ``examples/nlp/scripts/get_wkt2.sh``. After downloading and unzipping, the folder should include 3 files that look like this:
```
Comment (ekmb) [inline docs/sources/source/nlp/bert_pretraining.rst]: maybe also add that the user needs to specify the path to the data_dir to run this script

File: examples/nlp/language_modeling/bert_pretraining.py [hunk 3]
```diff
@@ -18,7 +18,8 @@
 To pretrain BERT on raw text dataset run
 python bert_pretraining.py \
 --amp_opt_level "O0" \
---data_dir data/lm/wikitext-2 \
+--train_data data/lm/wikitext-2/train.txt \
```
Comment (ekmb) [inline examples/nlp/language_modeling/bert_pretraining.py]: could you please change it to --train_data path_to/wikitext-2/train.txt and --eval_data path_to/wikitext-2/valid.txt ?
the user now can choose where to download the data, data/lm are gone

File: examples/nlp/language_modeling/bert_pretraining.py [hunk 4]
```diff
@@ -125,7 +127,8 @@
     help="maximum number of masked tokens to predict,\
                     needed when --preprocessed_data is specified",
 )
-parser.add_argument("--data_dir", default="data/lm/wikitext-2", type=str)
+parser.add_argument("--train_data", required=True, type=str, help="path to training dataset.")
+parser.add_argument("--eval_data", type=str, help="path to evaluation dataset.")
```
Comment (ekmb) [inline examples/nlp/language_modeling/bert_pretraining.py]: is it also required?

File: docs/sources/source/nlp/bert_pretraining.rst (hunk 1 above)
Comment (yzhang123) [reply docs/sources/source/nlp/bert_pretraining.rst]: this is for the on the fly processing case, which is done by BertPretrainingDataLayer.

File: examples/nlp/language_modeling/bert_pretraining.py (hunk 3 above)
Comment (yzhang123) [reply examples/nlp/language_modeling/bert_pretraining.py]: done

File: examples/nlp/language_modeling/bert_pretraining.py (hunk 4 above)
Comment (yzhang123) [reply examples/nlp/language_modeling/bert_pretraining.py]: yes, Done

File: docs/sources/source/nlp/bert_pretraining.rst (hunk 2 above)
Comment (yzhang123) [reply docs/sources/source/nlp/bert_pretraining.rst]: Done

**Pull request merged**